lib/flipper/adapters/http.rb

```diff
@@ -0,0 +1,197 @@
+require 'net/http'
+require 'json'
+require 'set'
+
+module Flipper
+  module Adapters
+    # class for handling http requests.
+    # Initialize with headers / basic_auth and use intance to make any requests
+    # headers and basic_auth will be sent in every request
+    # Request.new({ "X-Header" => "value" }, { "auth_username" => "auth_password" })
+    class Request
+      DEFAULT_HEADERS = {
+        'Content-Type' => 'application/json',
+        'Accept' => 'application/json',
+      }.freeze
+
+      def initialize(headers, basic_auth)
+        @headers = DEFAULT_HEADERS.merge(headers.to_h)
+        @basic_auth_username, @basic_auth_password = basic_auth.to_h.first
+      end
+
+      # Public: GET http request
+      def get(path)
+        uri = URI.parse(path)
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Get.new(uri.request_uri, @headers)
+        request.basic_auth(@basic_auth_username, @basic_auth_password) if basic_auth?
+        http.request(request)
+      end
+
+      # Public: POST http request
+      def post(path, data)
+        uri = URI.parse(path)
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Post.new(uri.request_uri, @headers)
+        request.body = data.to_json
+        request.basic_auth(@basic_auth_username, @basic_auth_password) if basic_auth?
+        http.request(request)
+      end
+
+      # Public: DELETE http request
+      def delete(path, data = {})
+        uri = URI.parse(path)
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Delete.new(uri.request_uri, @headers)
+        request.body = data.to_h.to_json
+        request.basic_auth(@basic_auth_username, @basic_auth_password) if basic_auth?
+        http.request(request)
+      end
```

Comment:
These three methods have very similar structure, but figured it better to go the boring easy to understand path over abstracting this into something like a request method that takes the http verb, etc.